src/test-jdk14/java/com/fasterxml/jackson/databind/RecordTest.java

```diff
@@ -52,7 +52,7 @@ public void testDeserializeSimpleRecord_DisableAnnotationIntrospector() throws I
         assertEquals(new SimpleRecord(123, "Bob"), value);
     }
 
-    record RecordOfRecord(SimpleRecord record) {
+    public record RecordOfRecord(SimpleRecord record) {
```

Comment:
_All test records should be public_

Why? (just for me to understand :))

Reply:
In Java 14 a record's canonical constructor is _always_ public ( it has the _public_ modifier ).
In Java 15 (and beyond) a record's canonical constructor inherits the access control modifier from that of the record declaration. 

So without the change to make the record declarations public, then in Java 15 the canonical constructor is package-private, and by default Jackson cannot locale it.   Now, this could be an issue or change with the default deserialisation policy in the version of the latest branch that I tested against.

Reply:
Thanks for the explanation

Reply:
We might want to try access of non-public Records too, for parity with POJOs. Although I don't know if it is possible to configure access for tests in a way that this is actually allowed (I assume module-info declarations are needed?).

Reply:
Adding tests to check for non-public Records would be good ( we have such in the JDK serialization unit tests ).

When running the existing RecordTest in my local environment with a JDK 15 early-access build I did see failures - since the record's canonical constructor was not public (it was package-private inherited from the record declaration). These failures may be because I was using the tip of jackson-databind where the default visibility of creators differs? Or maybe because of not explicitly setting the visibility checker in the test? I'm not sure.  Either way it should be possible to write tests where the record (and it's canonical constructor) are not accessible to Jackson ( package-private ), and ensure that whatever deserialization/visibility configuration work as expected.